raster-foundry/raster-foundry
# Implement project layer scenes and browse views

## Overview
Add project layer scene and browse views
Add layer parameter to /api/scenes, update datamodel to have inLayer property
Set hasNext correctly on /api/scenes if there are > 100 scenes
Use a sane default for the accepted query parameter on /api/project/{}/layer/{}/scenes endpoint

### Checklist

- ~Styleguide updated, if necessary~
- [x] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated [updated here](https://github.com/raster-foundry/raster-foundry-api-spec/pull/87)
- ~Symlinks from new migrations present or corrected for any new migrations~
- [x] Any content changes are properly templated using `BUILDCONFIG.APP_NAME` The styles are probably all very broken for non-RF core themes. @designmatty will need to do a pass over all of our styles before we do the full switch over to project layers on production
- ~Any new SQL strings have tests~

### Demo
![image](https://user-images.githubusercontent.com/4392704/52582889-425e5e00-2dfc-11e9-9ef1-dd572d5603a1.png)


### Notes
I'm pretty sure there are some missing features, but this has been worked on for long enough at this point that other issues should be made for them


## Testing Instructions
 * View an existing layer with scenes in it and verify that they load
* Verify that both options on the add imagery dropdown work
* Add a couple scenes to the first layer
* verify that you can preview scenes in the browse view by clicking on the icon
* Verify that the filter view works as expected (does not reload controllers etc)
* Verify that when you select a scene in the browse view, an option to add it to your layer apepars
* Verify that the checkbox is replaced with a loading icon while the scene is being added to the layer
* Verify that they show up in the browse view as already being in the layer
* Create another layer and browse with the same parameters / location, and verify that a third state exists for scenes which are in the project, but not the layer
* verify that you can still add scenes in a project to a second layer
* verify that the eye icon sticks the scene's geometry onto the map
* Verify that there are a reasonable number of requests when the browse view is initialized (2 vs 20)
* Verify that the download / delete buttons on the scenes view work as expected
* Verify that you can select and delete a set of scenes
* verify that when selecting scenes, the "select all visible" button works and interacts as expected (selects all scenes that are loaded into the UI)
* verify that you can view more than the first 100 scenes while browsing @jmorrison1847 this one's for you, the change should fix it for the old browse view as well


Closes #4537
